go/tasks/plugins/array/k8s/task.go

```diff
@@ -0,0 +1,233 @@
+	indexStr := strconv.Itoa(t.ChildIdx)
+	podName := formatSubTaskName(ctx, tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(), indexStr)
+	phaseInfo, err := CheckPodStatus(ctx, kubeClient,
+		k8sTypes.NamespacedName{
+			Name:      podName,
+			Namespace: tCtx.TaskExecutionMetadata().GetNamespace(),
+		})
+	if err != nil {
+		return MonitorError, errors2.Wrapf(ErrCheckPodStatus, err, "Failed to check pod status.")
+	}
+
+	logger.Debugf(ctx, "Monitor - phase info %v", phaseInfo)
+	if phaseInfo.Info() != nil {
+		t.LogLinks = append(t.LogLinks, phaseInfo.Info().Logs...)
+	}
+
+	if phaseInfo.Err() != nil {
+		t.MessageCollector.Collect(t.ChildIdx, phaseInfo.Err().String())
+	}
+
+	actualPhase := phaseInfo.Phase()
+	if phaseInfo.Phase().IsSuccess() {
+		originalIdx := arrayCore.CalculateOriginalIndex(t.ChildIdx, t.State.GetIndexesToCache())
+		actualPhase, err = array.CheckTaskOutput(ctx, dataStore, outputPrefix, baseOutputDataSandbox, t.ChildIdx, originalIdx)
+		if err != nil {
+			return MonitorError, err
+		}
+	}
+
+	t.NewArrayStatus.Detailed.SetItem(t.ChildIdx, bitarray.Item(actualPhase))
+	t.NewArrayStatus.Summary.Inc(actualPhase)
+
+	return MonitorSuccess, nil
+}
+
+func (t Task) Abort(ctx context.Context, tCtx core.TaskExecutionContext, kubeClient core.KubeClient) error {
+	indexStr := strconv.Itoa(t.ChildIdx)
+	podName := formatSubTaskName(ctx, tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(), indexStr)
+	pod := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       PodKind,
+			APIVersion: metav1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: tCtx.TaskExecutionMetadata().GetNamespace(),
+		},
+	}
+
+	err := kubeClient.GetClient().Delete(ctx, pod)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+
+			return nil
+		}
+		return err
+	}
+
+	return nil
+
+}
+
+func (t Task) Finalize(ctx context.Context, tCtx core.TaskExecutionContext, kubeClient core.KubeClient) error {
+	indexStr := strconv.Itoa(t.ChildIdx)
+	podName := formatSubTaskName(ctx, tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(), indexStr)
+
+	// Deallocate Resource
+	err := deallocateResource(ctx, tCtx, t.Config, t.ChildIdx)
+	if err != nil {
+		logger.Errorf(ctx, "Error releasing allocation token [%s] in Finalize [%s]", podName, err)
+		return err
+	}
+
+	return nil
+
+}
+
+func allocateResource(ctx context.Context, tCtx core.TaskExecutionContext, config *Config, podName string) (core.AllocationStatus, error) {
+	if !IsResourceConfigSet(config.ResourceConfig) {
+		return core.AllocationStatusGranted, nil
+	}
+
+	resourceNamespace := core.ResourceNamespace(config.ResourceConfig.PrimaryLabel)
+	resourceConstraintSpec := core.ResourceConstraintsSpec{
+		ProjectScopeResourceConstraint:   nil,
```

Comment:
why are these nil?

Reply:
Because we do not want any Project/Namespace based restriction today. We can add later if needed. I think it will not be used at all as resource quota will get a lot sooner. On top there is global limit as well.

Reply:
I'm confused though, don't we need to specify some constraint in order to leverage resource manager?

Reply:
or is the resourceNamespace that constraint and sufficient?